### PR TITLE
Feat: Object3D `attach()` accepts varargs

### PR DESCRIPTION
**Pull request by ycw**
This PR makes `attach()` variadic, as if `add()`

ex. attaches some children of one `Object3D` to another `Object3D`

```js
// without PR
one.children.slice(0,3).forEach(child=>another.attach(child))
// with PR
another.attach(...one.children.slice(0,3))
```

### CODE REVIEW

File: src/core/Object3D.js [hunk 1]
```diff
@@ -411,27 +406,58 @@ class Object3D extends EventDispatcher {
...
+
+			object.applyMatrix4( _m1 );
+
+			object.removeFromParent();
+			object.parent = this;
+			this.children.push( object );
+
+			object.updateWorldMatrix( false, true );
+
+			object.dispatchEvent( _addedEvent );
```
Comment (Mugen87) [inline src/core/Object3D.js:454]: Do you mind keep the usage of `add()` in this method? It should be sufficient if you just add the following code snippet at the top of the method:
```js
if ( arguments.length > 1 ) {

	for ( let i = 0; i < arguments.length; i ++ ) {

		this.attach( arguments[ i ] );

	}

	return this;

}
```

Comment (ycw) [reply src/core/Object3D.js:454]: if so, `added` will emit too early
```js
this.add( .. ) //  <<< emitted `added` already
object.updateWorldMatrix( false, true );
```
also, misleading error msg: `THREE.Object3D.add: ...` (should be `.attach`)

so, nope :)

Comment (Mugen87) [reply src/core/Object3D.js:454]: Ah okay, I wasn't aware of these issues. Nevertheless, it does not feel good that so much code is copy/pasted. I wonder if we can solve this differently.

**Comment by ycw** [comment]
[f2a658a](https://github.com/mrdoob/three.js/pull/26628/commits/f2a658ada2c9d8278471fd88ba61622940fd0383)
- add()/attach() calls _canAppend() to validate object
- _canAppend() can detect circular refs
- attach() supports varargs

**Comment by Mugen87** [comment]
Sorry to say this but the additions to `Object3D` still aren't satisfying. For now, I have the feeling it's better to not enhance `attach()`.

**Comment by ycw** [comment]
@Mugen87  maybe... using a sentinel? the idea:

```js
attach(...objects) {
   return this.add(objects, _keepTransform) // _keepTransform is a private symbol 
}

add(...objects) {
  // objects[1] is _keepTransform === delegated from attach()
}
```

this way `attach()` can delegate to `add()` elegantly... having same phases: 
1. first, validate objects (same err handling path)
2. then, process
3. finally, emit `added`

**Comment by mrdoob** [comment]
These methods are pretty delicate.

If someone loads a scene with thousands of objects it can get considerably slower.

**Comment by ycw** [comment]
thanks for your repsonse, i've created a new PR (26750) to clearly clarify my concerns, and explain why varidic attach() is an option, please review and discuss there.

**Pull request closed without merging**